webapp/tests/karma/unit/controllers/contacts-edit.js

```diff
@@ -91,11 +101,51 @@ describe('Contacts Edit controller', () => {
     let cancelCallback;
     spyState.params.id = 'id';
     globalActions.setCancelCallback.callsFake(func => cancelCallback = func);
+    lineageModelGenerator.contact.resolves({});
 
     createController();
     cancelCallback();
     chai.expect(spyState.go.callCount).to.equal(1);
     chai.expect(spyState.go.args[0]).to.deep.equal([ 'contacts.detail', { 'id': 'id' } ]);
   });
 
+  it('should select correct form for correct type', () => {
+    const contact = {
+      _id: 'contact_uuid',
+      type: 'person',
+      contact_type: 'whatever',
+    };
+    const form = {
+      _id: 'form:correct_edit_form',
+      internal_id: 'correct_edit_form',
+      type: 'form',
+    };
+
+    lineageModelGenerator.contact.resolves({ doc: contact });
+    spyState.params.id = contact._id;
+    contactTypes.getTypeId.returns('the correct type id');
+    contactTypes.get.resolves({ id: 'the correct type id', edit_form: 'correct_edit_form' });
+    db.get.resolves(form);
+
+    const controller = createController();
+    return tick().then(() => {
```

Comment:
Why do we need a delay here?

If this is just for the 3.10.x branch and we can do it without tick in angular 10 then it's fine.

Reply:
yea, the controller doesn't expose a promise that we can wait for until everything is done. 
In angular 10 it's much more elegant.